### PR TITLE
Removing dependency on moz_sql_parser

### PR DESCRIPTION
**Pull request by TGooch44**
Cleaning up the dependency on moz_sql_parser and instead relying on pyparsing directly.
cc: @danielcweeks @rdblue

### CODE REVIEW

File: python/iceberg/api/expressions/expression_parser.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          "!=": "neq",
+          "<>": "neq",
+          "neq": "neq",
+          "||": "or",
+          "or": "or",
+          "&&": "and",
+          "and": "and",
+          "in": "in",
+          "between": "between",
+          "is": "is"}
```
Comment (rdblue) [inline python/iceberg/api/expressions/expression_parser.py:89]: Is this derived from another parser or example? If so, we should make sure we note it so we can update the license documentation.

Comment (TGooch44) [reply python/iceberg/api/expressions/expression_parser.py:93]: Updated with some sourcing for the basic parsing grammar

File: python/tests/api/expressions/test_str_to_expr.py [hunk 2]
```diff
@@ -117,14 +117,32 @@ def test_ternary_condition():
 
 
 def test_precedence():
-    expected_expr = Expressions.and_(Expressions.or_(Expressions.equal("col_a", 1),
-                                                     Expressions.equal("col_b", 2)),
-                                     Expressions.equal("col_c", 3))
+    expected_expr = Expressions.or_(Expressions.equal("col_a", 1),
+                                    Expressions.and_(Expressions.equal("col_b", 2),
+                                                     Expressions.equal("col_c", 3)))
```
Comment (rdblue) [inline python/tests/api/expressions/test_str_to_expr.py:122]: Isn't it normal for OR to be lower precedence than AND? I think the original case was correct.

I tested this in Spark and Presto:
```sql
select * from rblue.test where id = 1 or id = 2 and data = 'b'
```
Presto:
```
filterPredicate = (("id" IN (BIGINT '1', BIGINT '2')) AND (("id" = BIGINT '1') OR ("data" = CAST('b' AS varchar))))
```
Spark:
```
filters=(id = 1 OR (id = 2 AND data = 'b'))
```

In both of them, AND has higher precedence than OR.

Comment (TGooch44) [reply python/tests/api/expressions/test_str_to_expr.py:122]: precedence in SQL is generally `()`, `AND`, `OR`.  Doesn't that mean the AND should be evaluated then the OR, which it looks like the newer case is doing?

Comment (rdblue) [reply python/tests/api/expressions/test_str_to_expr.py:122]: You're right! I got it backward and this is actually fixing the problem.

**Comment by rdblue** [comment]
+1 after @TGooch44 pointed out that precedence is fixed by this, not broken.

**Comment by rdblue** [comment]
Just thought of one more thing. Should we test NOT precedence as well?

**Comment by TGooch44** [comment]
I can add a test in a new PR since this was just merged

**Pull request merged**